pkg/init/cmd/rc.init/main.go

```diff
@@ -81,7 +81,7 @@ func mkdir(path string, perm os.FileMode) {
 
 // list of all enabled cgroups
 func cgroupList() []string {
-	list := []string{}
+	var list []string
```

Comment:
While we're updating this, I'm wondering if we should also optimise allocations, so instead of returning `list` (on error), just return `nil`, and when we successfully read the list, then initialise the slice with (`make([]string, 0, <size>)`.

For this function, something like:

```go
// list of all enabled cgroups
func cgroupList() []string {
	f, err := os.Open("/proc/cgroups")
	if err != nil {
		log.Printf("cannot open /proc/cgroups: %v", err)
		return nil
	}
	defer f.Close()
	reader := csv.NewReader(f)
	// tab delimited
	reader.Comma = '\t'
	// four fields
	reader.FieldsPerRecord = 4
	cgroups, err := reader.ReadAll()
	if err != nil {
		log.Printf("cannot parse /proc/cgroups: %v", err)
		return nil
	}
    list := make([]string, 0, len(cgroups))
	for _, cg := range cgroups {
		// see if enabled
		if cg[3] == "1" {
			list = append(list, cg[0])
		}
	}
	return list
}
```

Reply:
Yes, we could do that but that's a whole new level to find them all!